Won't start when I re-create deployment

Hi, I'm playing around with this and after deleting and recreating the etcd cluster I'm getting the following error logs:

```
time="2016-11-20T11:56:56Z" level=info msg="use in cluster client from k8s library" 
time="2016-11-20T11:56:56Z" level=info msg="finding existing clusters..." pkg=controller 
time="2016-11-20T11:56:56Z" level=info msg="starts running from watch version: 57524" pkg=controller 
time="2016-11-20T11:56:56Z" level=info msg="start watching at 57524" pkg=controller 
time="2016-11-20T11:57:01Z" level=warning msg="all etcd pods are dead. Trying to recover from a previous backup" cluster-name=etcd-cluster pkg=cluster 
time="2016-11-20T11:57:01Z" level=error msg="fail to do disaster recovery: no backup policy has been defined." cluster-name=etcd-cluster pkg=cluster 
time="2016-11-20T11:57:01Z" level=info msg="deleting cluster" cluster-name=etcd-cluster pkg=cluster 
panic: TODO: mark cluster dead if no backup for disaster recovery.

goroutine 34 [running]:
panic(0x1878380, 0xc420292ce0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/coreos/etcd-operator/pkg/cluster.(*Cluster).run(0xc4202df8f0, 0xc4203c5c80, 0xc4203c5910)
	/home/ubuntu/code/golang/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:183 +0x63a
created by github.com/coreos/etcd-operator/pkg/cluster.new
	/home/ubuntu/code/golang/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:119 +0x34e
```

No idea what's going on.

My manifest is as following:

```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: etcd-operator
spec:
  replicas: 1
  template:
    metadata:
      labels:
        name: etcd-operator
    spec:
      containers:
      - name: etcd-operator
        image: quay.io/coreos/etcd-operator
        env:
        - name: MY_POD_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
---
apiVersion: "coreos.com/v1"
kind: "EtcdCluster"
metadata:
  name: "etcd-cluster"
spec:
  size: 3
  version: "v3.1.0-alpha.1"
```

@hongchaodeng 

I agree we should not delete user created TPRs, existing Pods etc. when shutdown etcd operator. The existing etcd clusters should remain running no matter what happens to etcd operator.

But this still needs to be resolved.

```
panic: TODO: mark cluster dead if no backup for disaster recovery.

goroutine 34 [running]:
panic(0x1878380, 0xc420292ce0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/coreos/etcd-operator/pkg/cluster.(*Cluster).run(0xc4202df8f0, 0xc4203c5c80, 0xc4203c5910)
	/home/ubuntu/code/golang/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:183 +0x63a
created by github.com/coreos/etcd-operator/pkg/cluster.new
	/home/ubuntu/code/golang/src/github.com/coreos/etcd-operator/pkg/cluster/cluster.go:119 +0x34e
```

--- a/pkg/cluster/cluster.go
+++ b/pkg/cluster/cluster.go
@@ -267,7 +267,9 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 				err := c.disasterRecovery(nil)
 				if err != nil {
 					if err == errNoBackupExist {
-						panic("TODO: mark cluster dead if no backup for disaster recovery.")
+						c.logger.Error("cluster cannot be recovered: all members are dead and there is no backup")
+						c.logger.Warning("TODO: mark cluster dead if no backup for disaster recovery.")
+						return
 					}
 					c.logger.Errorf("fail to recover. Will retry later: %v", err)
 				}

